src/mrpro/data/IData.py

```diff
@@ -153,3 +153,15 @@ def from_dicom_folder(cls, foldername: str | Path, suffix: str | None = 'dcm') -
             raise ValueError(f'No dicom files with suffix {suffix} found in {foldername}')
 
         return cls.from_dicom_files(filenames=file_paths)
+
+    def __repr__(self):
+        """Representation method for IData class."""
+        try:
+            device = str(self.device)
+        except RuntimeError:
+            device = 'mixed'
+        out = (
+            f'IData with shape: {list(self.data.shape)!s} and dtype {self.data.dtype}\n'
```

Comment:
Can we use something like `type(self)` here instead of `IData` such that it also works for derived classes? Applies to the others too.

Reply:
`type(self)).__name__`